cpp/core/convert_core_json_to_marc.cc

```diff
@@ -0,0 +1,340 @@
+            "\t--create-unique-id-db: This flag has to be specified the first time this program will be executed only.\n"
+            "\t--ignore-unique-id-dups: If specified MARC records will be created for unique ID's which we have encountered\n"
+            "\t                         before.  The unique ID database will still be updated.\n"
+            "\t--extract-and-count-issns-only: Generates stats on the frequency of ISSN's in the JSON input and does not generate any \n"
+            "\t                                MARC output files.  This requires the existence of the \"magic\" \"ISSN\" config file entry!\n"
+            "\tunmapped_issn_list (output): Here we list the ISSN's for which we have no entry in issns_to_journaltitles_and_ppns.map,\n"
+            "\t                             required unless --extract-and-count-issns-only was specified!\n"
+            "\tmarc_outpt: required unless --extract-and-count-issns-only was specified!\n\n");
+}
+
+
+// Parses an input file that has three (the last component may be empty) parts per line that are colon-separated.
+// Embedded colons may be backslash escaped.
+void LoadISSNsToJournalTitlesPPNsAndISSNsMap(
+    std::unordered_map<std::string, JournalTitlePPNAndOnlineISSN> * const issns_to_journal_titles_ppns_and_issns_map)
+{
+    const std::string MAP_FILE_PATH(UBTools::GetTuelibPath() + "issns_to_journaltitles_and_ppns.map");
+    const auto input(FileUtil::OpenInputFileOrDie(MAP_FILE_PATH));
+
+    unsigned line_no(0);
+    while (not input->eof()) {
+        ++line_no;
+        const std::string line(input->getline());
+        if (line.empty())
+            continue;
+        std::vector<std::string> parts(SplitLineOnColons(line));
+        if (parts.size() != 4 or parts[0].empty() or parts[1].empty()) // ISSN and titles are required, PPN's and online ISSN's are optional.
+            LOG_ERROR("malformed line #" + std::to_string(line_no) + " in \"" + MAP_FILE_PATH + "\"!");
+        issns_to_journal_titles_ppns_and_issns_map->emplace(parts[0], JournalTitlePPNAndOnlineISSN(parts[1], parts[2],
+                                                                                                   parts[3]));
+    }
+
+    LOG_INFO("Loaded " + std::to_string(issns_to_journal_titles_ppns_and_issns_map->size())
+             + " mappings from print ISSN's to online ISSN's, PPN's and journal titles.");
+}
+
+
+MARC::Record::BibliographicLevel MapTypeStringToBibliographicLevel(const std::string &item_type) {
+    if (item_type == "Book item")
+        return MARC::Record::MONOGRAPH_OR_ITEM;
+    else if (item_type == "Book chapter")
+        return MARC::Record::MONOGRAPHIC_COMPONENT_PART;
+    else if (item_type == "Article")
+        return MARC::Record::SERIAL_COMPONENT_PART;
+    else {
+        LOG_WARNING("unkown item type: " + item_type);
+        return MARC::Record::UNDEFINED;
+    }
+}
+
+
+std::string NormalizeAuthorName(const std::string &author_name) {
```

Comment:
Can we re-use the Zotaut logic for normalizing author names here?

Reply:
We also have NormaliseAuthorName in ControlNumberGuesser.cc which does some of the same work and some additional stuff.  Part of that could also be extracted into a common function.